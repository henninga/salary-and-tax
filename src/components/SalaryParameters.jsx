import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import { reduxForm } from 'redux-form';
import { inputHasChanged } from '../actions';
import { Button, Checkbox, FormGroup, InputWithLabel, Section } from './Ui';


export const SalaryParametersComponent = ({handleSubmit, submitting}) => (
  <Section header="Salary input">
    <form onSubmit={handleSubmit}>
      <FormGroup header="Turnover">
        <InputWithLabel name="hoursPerYear" label="Hours per year" />
        <InputWithLabel name="hourRate" label="Hour rate" />
      </FormGroup>

      <FormGroup header="Subcontractor">
        <InputWithLabel name="cut" label="Cut (%)" />
      </FormGroup>

      <FormGroup header="Employer Fee">
        <Checkbox name="includeEmployerFee" label="Include in calculation" />
        <InputWithLabel name="employerFeeRate" label="(%)" />
      </FormGroup>

      <FormGroup header="Vacation savings">
        <Checkbox name="includeVacationSavings" label="Include in calculation" />
        <InputWithLabel name="vacationSavingsRate" label="(%)" />
      </FormGroup>

      <FormGroup header="Pension">
        <Checkbox name="includePension" label="Include in calculation" />
        <InputWithLabel name="pensionOneToSixRate" label="Percentage (1G-6G)" />
        <InputWithLabel name="pensionSixToTwelveRate" label="Percentage (6G-12G)" />
      </FormGroup>

      <Button type="submit" disabled={submitting}>Calculate earnings</Button>
    </form>
  </Section>);

SalaryParametersComponent.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  submitting: PropTypes.bool.isRequired,
};

export const SalaryParametersForm = reduxForm({form: 'calculations'})(SalaryParametersComponent);

const mapStateToProps = ({calculations: {data}}) => ({
  initialValues: data,
});

const mapDispatchToProps = (dispatch) => ({
  onSubmit: (values) => dispatch(inputHasChanged(values)),
});

export const SalaryParameters = connect(mapStateToProps, mapDispatchToProps)(SalaryParametersForm);
export default SalaryParameters;
