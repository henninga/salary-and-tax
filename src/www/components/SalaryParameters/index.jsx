import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import { reduxForm } from 'redux-form';
import { inputHasChanged } from '../../actions';
import { InputWithLabel, Section } from '../Ui';
import { getGridClass } from '../Ui/Grid';


export const SalaryParametersComponent = ({handleSubmit, submitForm, submitting}) => (
  <form onSubmit={handleSubmit(submitForm)} onChange={submitForm} className={getGridClass(6)}>
    <Section header="Income" gridValue={6}>
      <InputWithLabel fieldName="hoursPerYear" labelText="Hours per year" />
      <InputWithLabel fieldName="hourRate" labelText="Hour rate" />
    </Section>

    <Section header="Vacation savings" gridValue={6}>
      <InputWithLabel fieldName="vacationSavingsRate" labelText="(%)" />
      <InputWithLabel fieldName="includeVacationSavings" labelText="Include in calculation" type="checkbox" component="input" />
    </Section>

    <Section header="Subcontractor" gridValue={6}>
      <InputWithLabel fieldName="cut" labelText="Cut (%)" />
    </Section>

    <Section header="Employer Fee" gridValue={6}>
      <InputWithLabel fieldName="employerFeeRate" labelText="(%)" />
      <InputWithLabel fieldName="includeEmployerFee" labelText="Include in calculation" type="checkbox" component="input" />
    </Section>

    <Section header="Pension" gridValue={6}>
      <InputWithLabel fieldName="pensionOneToSixRate" labelText="Percentage (1G-6G)" />
      <InputWithLabel fieldName="includePension" labelText="Include in calculation" type="checkbox" component="input" />
      <InputWithLabel fieldName="pensionSixToTwelveRate" labelText="Percentage (6G-12G)" />
    </Section>

    <button type="submit" disabled={submitting} className={getGridClass(6)}>Calculate earnings</button>
  </form>);

SalaryParametersComponent.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  submitForm: PropTypes.func.isRequired,
  submitting: PropTypes.bool.isRequired,
};

const SalaryParametersForm = reduxForm({form: 'calculations'})(SalaryParametersComponent);

const mapStateToProps = ({calculations: {data}}) => ({
  initialValues: data,

});

const mapDispatchToProps = (dispatch) => ({
  submitForm: (values) => dispatch(inputHasChanged(values)),
});


export const SalaryParameters = connect(mapStateToProps, mapDispatchToProps)(SalaryParametersForm);
export default SalaryParameters;
