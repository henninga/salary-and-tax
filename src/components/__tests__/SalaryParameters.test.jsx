import mockComponent from 'identity-obj-proxy'
import React from 'react';
import renderer from 'react-test-renderer';
import { SalaryParametersComponent } from '../SalaryParameters';


const handleSubmit = fn => fn();
const submitForm = jest.fn();

jest.mock('../Ui', () => mockComponent);

describe('SalaryParametersComponent', () => {
  it('should render', () => {
    const tree = renderer.create(<SalaryParametersComponent
      handleSubmit={handleSubmit}
      submitForm={submitForm}
      submitting={false}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
