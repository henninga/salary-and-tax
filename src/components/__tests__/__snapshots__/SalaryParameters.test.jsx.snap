// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SalaryParametersComponent should render 1`] = `
<Section
  expanded={true}
  header="Parameters"
>
  <form
    onSubmit={[Function]}
  >
    <FormGroup
      header="Turnover"
    >
      <InputWithLabel
        label="Hours per year"
        name="hoursPerYear"
      />
      <InputWithLabel
        label="Hour rate"
        name="hourRate"
      />
    </FormGroup>
    <FormGroup
      header="Subcontractor"
    >
      <InputWithLabel
        label="Cut (%)"
        name="cut"
      />
    </FormGroup>
    <FormGroup
      header="Employer Fee"
    >
      <Checkbox
        label="Include in calculation"
        name="includeEmployerFee"
      />
      <InputWithLabel
        label="(%)"
        name="employerFeeRate"
      />
    </FormGroup>
    <FormGroup
      header="Vacation savings"
    >
      <Checkbox
        label="Include in calculation"
        name="includeVacationSavings"
      />
      <InputWithLabel
        label="(%)"
        name="vacationSavingsRate"
      />
    </FormGroup>
    <FormGroup
      header="Pension"
    >
      <Checkbox
        label="Include in calculation"
        name="includePension"
      />
      <InputWithLabel
        label="(%) (1G-6G)"
        name="pensionOneToSixRate"
      />
      <InputWithLabel
        label="(%) (6G-12G)"
        name="pensionSixToTwelveRate"
      />
    </FormGroup>
    <FormGroup
      header="Language"
    >
      <RadioGroup
        label="Language"
        name="locale"
      >
        <FormControlLabel
          control={<Radio />}
          label="English"
          value="en"
        />
        <FormControlLabel
          control={<Radio />}
          label="Norsk"
          value="nb"
        />
      </RadioGroup>
    </FormGroup>
    <Button
      disabled={false}
      type="submit"
    >
      Calculate salary
    </Button>
  </form>
</Section>
`;
